src/main/java/com/maddyhome/idea/vim/extension/nerdtree/Mappings.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2003-2025 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.extension.nerdtree
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.ui.treeStructure.Tree
+import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.api.injector
+import com.maddyhome.idea.vim.key.KeyStrokeTrie
+import com.maddyhome.idea.vim.key.add
+import com.maddyhome.idea.vim.newapi.vim
+import com.maddyhome.idea.vim.vimscript.model.datatypes.VimString
+import javax.swing.KeyStroke
+
+/**
+ * Maps key sequences to NERDTree actions using KeyStrokeTrie for efficient lookup.
+ *
+ * @constructor Creates an empty trie for key mappings
+ * @param name The name of the KeyStrokeTrie instance (for debug purposes)
+ */
+internal class Mappings(name: String) {
+  private val trie = KeyStrokeTrie<Action>(name)
+  val getAction = trie::getData
+
+  private val _keyStrokes = mutableSetOf<KeyStroke>()
+  val keyStrokes: Set<KeyStroke> get() = _keyStrokes.toSet()
+
+  fun register(variable: String, defaultMapping: String, action: Action) {
+    val variableValue = VimPlugin.getVariableService().getGlobalVariableValue(variable)
+    val mapping = if (variableValue is VimString) {
+      variableValue.value
+    } else {
+      defaultMapping
+    }
+    register(mapping, action)
+  }
+
+  fun register(mapping: String, action: Action) {
+    trie.add(mapping, action)
+    _keyStrokes.addAll(injector.parser.parseKeys(mapping))
+  }
+
+  /**
+   * Defines the actual behavior of actions in NERDTree
+   */
+  class Action(val action: (AnActionEvent, Tree) -> Unit) {
+    companion object {
+      private fun executeAction(event: AnActionEvent, id: String) =
+        NerdTree.Util.callAction(null, id, event.dataContext.vim)
+
+      /**
+       * Creates an Action that executes a series of IntelliJ actions identified by their IDs.
+       *
+       * @param ids A variable number of strings representing the IDs of the actions to execute.
+       * @return An Action that runs the specified actions when triggered.
+       */
+      fun ij(vararg ids: String): Action = Action { event, _ -> ids.forEach { id -> executeAction(event, id) } }
+    }
```

Comment:
Calling actions one by one is a complicated topic. The action initially called from the EDT can easily switch to the background thread, postpone execution and do almost anything. I see, there is a single use of multiple actions, is it possible to replace it with code alternative?